README.md

```diff
@@ -27,6 +25,53 @@ messages from a mediator and being able to asynchronously receive messages over
 a transport, such an agent could use a separate service that implements the
 active polling and then forwards/relays the retrieved messages to it.
 
+
+
+# Quickstart
+
+
+Clone the Proxy Mediator and prepare a virtual environment:
+```sh
+$ git clone https://github.com/Indicio-tech/proxy-mediator.git
+$ cd proxy-mediator
+$ python3 -m venv env
+$ source env/bin/activate
+```
+
+## To run the demo setup:
+  - `docker-compose up --build`
+
+
+## Services started by the above command:
+- **Proxy**: The mediator inside the firewall which connects to the external mediator.
+- **Agent**: The agent that is sending and receiving messages through the mediators.
+- **Mediator-tunnel**: The tunnel through which the external mediator transfers information.
+- **Reverse-proxy**: The service that redirects traffic from one endpoint to another.
+- **Mediator**: The external mediator that connects to the internal mediator from outside of the firewall.
+- **Setup**: Builds the context and sets the environment variables.
+
+  - The MessageRetriever retrieves messages by opening a websocket connection and continuously polling for messages using a trust ping with `response_requested` set to false while the websocket connection is open.
+
+
+# Setup
+
+Steps to set up a connection to the mediators:
+1. Retrieve the invitation from the external mediator
+    - Use the `acapy_client` to create an invitation request for the external mediator and return the invitation url.
```

Comment:
Usage of the `acapy-client` library is also an implementation detail we ought to "hide." I would include the specific ACA-Py endpoints being called.